### PR TITLE
ESlint + code format

### PR DESCRIPTION
**Pull request by nathan-de-pachtere**
Add ESLint to enforce Vue common code linting rules. 

@rahul-vashishtha This merge request is just to gather discussion and have a base to work on.

TODO

- [x] Install ESLINT
- [x] Define the rules we want to remove
- [x] Define which Prettier rules we want
- [x] Format the entire repo to align 😨
- [x] Enjoy no more formatting/linting messages in code reviews and focus on other important stuff

### CODE REVIEW

File: prettier.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default {
+  semi: true,
+  printWidth: 100,
+  tabWidth: 2,
+  singleAttributePerLine: true,
```
Comment (nathan-de-pachtere) [inline prettier.config.js:5]: Amazing for readability

image
image

File: eslint.conf.js
Comment (nathan-de-pachtere) [inline eslint.conf.js:1]: This entire file is to refactor to the needs, just copy /paste/clean from my monorepo

**Comment by nathan-de-pachtere** [comment]
@rahul-vashishtha Asked changes done ;)

File: eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      "@typescript-eslint/no-empty-object-type": [
+        "error",
+        { allowInterfaces: "with-single-extends" },
+      ],
+      "@typescript-eslint/no-unused-vars": "off",
+      "check-file/filename-naming-convention": [
+        "error",
+        {
+          "**/*.md": "KEBAB_CASE",
+          "**/!(*.config).{js,ts}": "PASCAL_CASE",
```
Comment (rahul-vashishtha) [inline eslint.config.js:54]: Pascal Case is only for vue files under `components` folder.

js and ts can be kebab case.

**Review by rahul-vashishtha — CHANGES_REQUESTED** [review]
Just a minor change required. All set then.

File: eslint.config.js (hunk 2 above)
Comment (nathan-de-pachtere) [reply eslint.config.js]: Done

**Pull request merged**